src/transformStateWithClones.js

```diff
@@ -1,13 +1,27 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const currentState = Object.assign({}, state);
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const { type, extraData, keysToRemove } = actions[i];
+
+    if (type === 'addProperties') {
+      Object.assign(currentState, extraData);
+    } else if (type === 'removeProperties') {
+      keysToRemove.forEach((key) => {
+        delete currentState[key];
+      });
+    } else {
+      for (const key in currentState) {
+        delete currentState[key];
+      }
+    }
```

Comment:
[CODE STYLE]: Use a switch statement when you have a limited number of conditions, as it makes the code cleaner and easier to read. Also, ensure to include a default case for error handling.